paymentsheet/src/main/java/com/stripe/android/paymentsheet/forms/FormViewModel.kt

```diff
@@ -122,19 +120,27 @@ internal class FormViewModel @Inject internal constructor(
 
     val completeFormValues =
         CompleteFormFieldValueFilter(
-            combine(
-                elements.map {
-                    it.getFormFieldValueFlow()
-                }
-            ) {
-                it.toList().flatten().toMap()
-            },
+            currentFieldValues().map { it.toMap() },
             hiddenIdentifiers,
             showingMandate,
             userRequestedReuse,
             defaultValuesToInclude,
         ).filterFlow()
 
+    private fun currentFieldValues(): Flow<List<Pair<IdentifierSpec, FormFieldEntry>>> {
```

Comment:
This is the bulk of the fix, ensuring we emit when elements is empty (which is now happening).